<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
<class>MainWindow</class>
<widget class="QMainWindow" name="MainWindow">
    <property name="geometry">
    <rect>
        <x>0</x>
        <y>0</y>
        <width>800</width>
        <height>600</height>
    </rect>
    </property>
    <property name="windowTitle">
    <string>MainWindow</string>
    </property>
    <widget class="QWidget" name="centralwidget">
    <layout class="QGridLayout" name="gridLayout">
        <!-- Display -->
        <item row="0" column="0" colspan="5">
        <widget class="QLineEdit" name="display">
        <property name="readOnly">
        <bool>true</bool>
        </property>
        <property name="alignment">
        <set>Qt::AlignRight</set>
        </property>
        <property name="text">
        <string>0</string>
        </property>
        </widget>
        </item>

        <!-- Memory Buttons Row -->
        <item row="1" column="0">
        <widget class="QPushButton" name="button_deg">
        <property name="text">
        <string>DEG</string>
        </property>
        </widget>
        </item>
        <item row="1" column="1">
        <widget class="QPushButton" name="button_fe">
        <property name="text">
        <string>F-E</string>
        </property>
        </widget>
        </item>
        <item row="1" column="2">
        <widget class="QPushButton" name="button_mc">
        <property name="text">
        <string>MC</string>
        </property>
        </widget>
        </item>
        <item row="1" column="3">
        <widget class="QPushButton" name="button_mr">
        <property name="text">
        <string>MR</string>
        </property>
        </widget>
        </item>
        <item row="1" column="4">
        <widget class="QPushButton" name="button_mplus">
        <property name="text">
        <string>M+</string>
        </property>
        </widget>
        </item>

        <!-- Basic Buttons -->
        <item row="2" column="0">
        <widget class="QPushButton" name="button_clear">
        <property name="text">
        <string>C</string>
        </property>
        </widget>
        </item>
        <item row="2" column="1">
        <widget class="QPushButton" name="button_backspace">
        <property name="text">
        <string>⌫</string>
        </property>
        </widget>
        </item>
        <item row="2" column="2">
        <widget class="QPushButton" name="button_sign">
        <property name="text">
        <string>±</string>
        </property>
        </widget>
        </item>
        <item row="2" column="3">
        <widget class="QPushButton" name="button_divide">
        <property name="text">
        <string>÷</string>
        </property>
        </widget>
        </item>

        <!-- 2ndボタン追加（row=2→row=7に変更） -->
        <item row="7" column="0">
        <widget class="QPushButton" name="button_2nd">
        <property name="text">
        <string>2nd</string>
        </property>
        </widget>
        </item>
        <!-- 三角関数コンボボックス追加（row=2→row=7, column=1に変更） -->
        <item row="7" column="1">
        <widget class="QComboBox" name="combo_trig">
        <property name="editable">
        <bool>false</bool>
        </property>
        <item><property name="text"><string>2nd</string></property></item>
        <item><property name="text"><string>sin</string></property></item>
        <item><property name="text"><string>cos</string></property></item>
        <item><property name="text"><string>tan</string></property></item>
        <item><property name="text"><string>hyp</string></property></item>
        <item><property name="text"><string>sec</string></property></item>
        <item><property name="text"><string>csc</string></property></item>
        <item><property name="text"><string>cot</string></property></item>
        </widget>
        </item>
        <!-- 関数コンボボックス追加（row=2→row=7, column=2に変更） -->
        <item row="7" column="2">
        <widget class="QComboBox" name="combo_func">
        <property name="editable">
        <bool>false</bool>
        </property>
        <item><property name="text"><string>abs</string></property></item>
        <item><property name="text"><string>floor</string></property></item>
        <item><property name="text"><string>ceil</string></property></item>
        <item><property name="text"><string>rand</string></property></item>
        <item><property name="text"><string>→dms</string></property></item>
        <item><property name="text"><string>→deg</string></property></item>
        </widget>
        </item>
        <!-- 主要関数ボタン群追加（例、row=7, column=3以降に変更） -->
        <item row="7" column="3">
        <widget class="QPushButton" name="button_pow2">
        <property name="text"><string>x²</string></property>
        </widget>
        </item>
        <item row="7" column="4">
        <widget class="QPushButton" name="button_sqrt">
        <property name="text"><string>√x</string></property>
        </widget>
        </item>
        <item row="8" column="3">
        <widget class="QPushButton" name="button_powxy">
        <property name="text"><string>xʸ</string></property>
        </widget>
        </item>
        <item row="8" column="4">
        <widget class="QPushButton" name="button_pow10">
        <property name="text"><string>10ˣ</string></property>
        </widget>
        </item>
        <item row="8" column="2">
        <widget class="QPushButton" name="button_log">
        <property name="text"><string>log</string></property>
        </widget>
        </item>
        <item row="8" column="1">
        <widget class="QPushButton" name="button_exp">
        <property name="text"><string>exp</string></property>
        </widget>
        </item>

        <!-- Digit Buttons 7-9 -->
        <item row="3" column="0">
        <widget class="QPushButton" name="button_7">
        <property name="text">
        <string>7</string>
        </property>
        </widget>
        </item>
        <item row="3" column="1">
        <widget class="QPushButton" name="button_8">
        <property name="text">
        <string>8</string>
        </property>
        </widget>
        </item>
        <item row="3" column="2">
        <widget class="QPushButton" name="button_9">
        <property name="text">
        <string>9</string>
        </property>
        </widget>
        </item>
        <item row="3" column="3">
        <widget class="QPushButton" name="button_multiply">
        <property name="text">
        <string>×</string>
        </property>
        </widget>
        </item>

        <!-- Digit Buttons 4-6 -->
        <item row="4" column="0">
        <widget class="QPushButton" name="button_4">
        <property name="text">
        <string>4</string>
        </property>
        </widget>
        </item>
        <item row="4" column="1">
        <widget class="QPushButton" name="button_5">
        <property name="text">
        <string>5</string>
        </property>
        </widget>
        </item>
        <item row="4" column="2">
        <widget class="QPushButton" name="button_6">
        <property name="text">
        <string>6</string>
        </property>
        </widget>
        </item>
        <item row="4" column="3">
        <widget class="QPushButton" name="button_minus">
        <property name="text">
        <string>-</string>
        </property>
        </widget>
        </item>

        <!-- Digit Buttons 1-3 -->
        <item row="5" column="0">
        <widget class="QPushButton" name="button_1">
        <property name="text">
        <string>1</string>
        </property>
        </widget>
        </item>
        <item row="5" column="1">
        <widget class="QPushButton" name="button_2">
        <property name="text">
        <string>2</string>
        </property>
        </widget>
        </item>
        <item row="5" column="2">
        <widget class="QPushButton" name="button_3">
        <property name="text">
        <string>3</string>
        </property>
        </widget>
        </item>
        <item row="5" column="3">
        <widget class="QPushButton" name="button_plus">
        <property name="text">
        <string>+</string>
        </property>
        </widget>
        </item>

        <!-- Digit Button 0 and Decimal -->
        <item row="6" column="0" colspan="2">
        <widget class="QPushButton" name="button_0">
        <property name="text">
        <string>0</string>
        </property>
        </widget>
        </item>
        <item row="6" column="2">
        <widget class="QPushButton" name="button_dot">
        <property name="text">
        <string>.</string>
        </property>
        </widget>
        </item>
        <item row="6" column="3">
        <widget class="QPushButton" name="button_equal">
        <property name="text">
        <string>=</string>
        </property>
        </widget>
        </item>
    </layout>
    </widget>
    <widget class="QMenuBar" name="menubar">
    <property name="geometry">
        <rect>
        <x>0</x>
        <y>0</y>
        <width>800</width>
        <height>22</height>
        </rect>
    </property>
    </widget>
    <widget class="QStatusBar" name="statusbar"/>
    </widget>
    <resources/>
    <connections/>
</ui>
